### PR TITLE
Add `Dialog` component

### PR DESCRIPTION
**Pull request by michaeltaranto**
Add `Dialog` component

Follows the [WAI Aria Dialog (Modal) Pattern](https://www.w3.org/TR/wai-aria-practices-1.2/#dialog_modal).

**EXAMPLE USAGE:**

```jsx
<Fragment>
  <Actions>
    <Button onClick={() => setOpen(true)}>Open dialog</Button>
  </Actions>
  <Dialog
    title="Dialog Example"
    id={id}
    open={open}
    onClose={setOpen}
  >
    <Placeholder height={100} width="100%" />
  </Dialog>
</Fragment>
```

See [the documentation](https://braid-design-system--0284a800a427955e7933c039ecf3f5eb9fd3fee2.surge.sh/components/Dialog) for an exhaustive list of features.

#### Developer Notes
- Moved from `isloated-scroll` to `react-remove-scroll`, now supports touch devices and keyboard locking.
- Upgrade the `testing-library` eco-system for better focus testing apis.

### CODE REVIEW

File: .changeset/strange-coats-kick.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+'braid-design-system': minor
+---
+
+Add `Dialog` component
```
Comment (markdalgleish) [inline .changeset/strange-coats-kick.md:5]: Needs the `added` front matter.

File: lib/components/Dialog/Dialog.docs.tsx [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+  Strong,
+} from '../';
+import {
+  Placeholder,
+  Dialog as PlayroomDialog,
+} from '../../playroom/components';
+
+const docs: ComponentDocs = {
+  category: 'Layout',
+  screenshotWidths: [320],
```
Comment (markdalgleish) [inline lib/components/Dialog/Dialog.docs.tsx:22]: Is there any point screenshotting this?

File: lib/components/Dialog/Dialog.docs.tsx [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+  description: (
+    <Stack space="large">
+      <Text>
+        Follows the{' '}
+        <TextLink href="https://www.w3.org/TR/wai-aria-practices-1.2/#dialog_modal">
+          WAI-ARIA Dialog (Modal) Pattern.
+        </TextLink>
+      </Text>
+      <Text>
+        The Dialog component provides a way to focus the users attention on a
```
Comment (markdalgleish) [inline lib/components/Dialog/Dialog.docs.tsx:32]:
```suggestion
        The Dialog component provides a way to focus the user's attention on a
```

File: lib/components/Dialog/Dialog.docs.tsx [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+                  </Fragment>
+                }
+              />
+            </Dialog>
+          </Fragment>
+        );
+      },
+    },
+    {
+      label: 'With additional description',
```
Comment (markdalgleish) [inline lib/components/Dialog/Dialog.docs.tsx:279]: Should this example be higher up, before all the different sizes?

File: lib/components/Dialog/Dialog.docs.tsx (hunk 2 above)
Comment (michaeltaranto) [reply lib/components/Dialog/Dialog.docs.tsx:22]: Im abstracting out a DialogCard internally and going to screenshot that. Its too important to not screenshot, but these in their current state are of no use.

File: lib/components/Dialog/DialogCard.treat.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import { style } from 'sku/treat';
```
Comment (mattcompiles) [inline lib/components/Dialog/DialogCard.treat.ts:1]: How would you feel about putting all the styles in `Dialog.treat` rather than two separate files?

**Review by markdalgleish — APPROVED** [review]
🚀

**Pull request merged**